provider/cluster_rosa_classic_resource.go

```diff
@@ -1990,3 +2055,50 @@ func adminCredsValidators() []tfsdk.AttributeValidator {
 		},
 	}
 }
+
+func validatePrivateHostedZone(clusterState *ClusterRosaClassicState) error {
+	if clusterState.PrivateHostedZone == nil {
+		// Nothing to validate.
+		return nil
+	}
+	// validate ID and ARN are not empty
+	if common.IsStringAttributeEmpty(clusterState.PrivateHostedZone.ID) || common.IsStringAttributeEmpty(clusterState.PrivateHostedZone.RoleARN) {
+		return fmt.Errorf("Invalid configuration. 'private_hosted_zone.id' and 'private_hosted_zone.arn' are required")
+	}
+	// Validate running in STS mode
+	if clusterState.Sts == nil {
+		return fmt.Errorf("Invalid configuration. 'private_hosted_zone' requires 'sts' configueration")
+	}
+	// Validate subnets exists
+	if len(clusterState.AWSSubnetIDs.Elems) <= 0 {
+		return fmt.Errorf("Invalid configuration. 'private_hosted_zone' requires 'aws_subnet_ids' configueration")
+	}
+	// Validate availabilityZones exists
+	if len(clusterState.AvailabilityZones.Elems) <= 0 {
+		return fmt.Errorf("Invalid configuration. 'private_hosted_zone' requires 'aws_subnet_ids' configueration")
+	}
+	// Validate BaseDomain
+	if common.IsStringAttributeEmpty(clusterState.BaseDNSDomain) {
+		return fmt.Errorf("Invalid configuration. 'private_hosted_zone' requires 'base_dns_domain' configueration")
+	}
+	return nil
+}
+
+// Place holder until the v2 refactoring
+func privateHZValidators() tfsdk.AttributeValidator {
+	return &common.AttributeValidator{
+		Desc: "Validate private_hosted_zone",
+		Validator: func(ctx context.Context, req tfsdk.ValidateAttributeRequest, resp *tfsdk.ValidateAttributeResponse) {
+			var clusterState *ClusterRosaClassicState
```

Comment:
I'm not sure I will have the connection object in the SDK v2 in that validator, BUT it seems to be from type list so we will not have a validator in the plan step any way :D

Reply:
This should be a map. With 2 strings. Right? Not a list.

Reply:
No 
Map is only for "primitive" attributes....

Reply:
We will change it when rebase on new version